src/event.rs

```diff
@@ -0,0 +1,587 @@
+					ldk_events::PaymentPurpose::InvoicePayment {
+						payment_preimage,
+						payment_secret,
+						..
+					} => (*payment_preimage, Some(*payment_secret)),
+					ldk_events::PaymentPurpose::SpontaneousPayment(preimage) => {
+						(Some(*preimage), None)
+					}
+				};
+				let mut payments = self.inbound_payments.lock().unwrap();
+				match payments.entry(*payment_hash) {
+					hash_map::Entry::Occupied(mut e) => {
+						let payment = e.get_mut();
+						payment.status = PaymentStatus::Succeeded;
+						payment.preimage = payment_preimage;
+						payment.secret = payment_secret;
+					}
+					hash_map::Entry::Vacant(e) => {
+						e.insert(PaymentInfo {
+							preimage: payment_preimage,
+							secret: payment_secret,
+							status: PaymentStatus::Succeeded,
+							amount_msat: Some(*amount_msat),
+						});
+					}
+				}
+			}
+			ldk_events::Event::PaymentSent {
+				payment_preimage,
+				payment_hash,
+				fee_paid_msat,
+				..
+			} => {
+				let mut payments = self.outbound_payments.lock().unwrap();
+				for (hash, payment) in payments.iter_mut() {
+					if *hash == *payment_hash {
+						payment.preimage = Some(*payment_preimage);
+						payment.status = PaymentStatus::Succeeded;
+						log_info!(
+							self.logger,
+							"Successfully sent payment of {} millisatoshis{} from \
+								 payment hash {:?} with preimage {:?}",
+							payment.amount_msat.unwrap(),
+							if let Some(fee) = fee_paid_msat {
+								format!(" (fee {} msat)", fee)
+							} else {
+								"".to_string()
+							},
+							hex_utils::to_string(&payment_hash.0),
+							hex_utils::to_string(&payment_preimage.0)
+						);
+					}
+				}
+				self.event_queue
+					.add_event(LdkLiteEvent::PaymentSuccessful(PaymentSuccessfulEvent {
+						payment_hash: *payment_hash,
+					}))
+					.unwrap();
+			}
+			ldk_events::Event::PaymentFailed { payment_hash, .. } => {
+				log_info!(
+					self.logger,
+					"Failed to send payment to payment hash {:?}: exhausted payment retry attempts",
+					hex_utils::to_string(&payment_hash.0)
+				);
+
+				let mut payments = self.outbound_payments.lock().unwrap();
+				if payments.contains_key(&payment_hash) {
+					let payment = payments.get_mut(&payment_hash).unwrap();
+					payment.status = PaymentStatus::Failed;
+				}
+				self.event_queue
+					.add_event(LdkLiteEvent::PaymentFailed(PaymentFailedEvent {
+						payment_hash: *payment_hash,
+					}))
+					.unwrap();
+			}
+
+			ldk_events::Event::PaymentPathSuccessful { .. } => {}
+			ldk_events::Event::PaymentPathFailed { .. } => {}
+			ldk_events::Event::ProbeSuccessful { .. } => {}
+			ldk_events::Event::ProbeFailed { .. } => {}
+			ldk_events::Event::HTLCHandlingFailed { .. } => {}
+			ldk_events::Event::PendingHTLCsForwardable { time_forwardable } => {
+				let forwarding_channel_manager = self.channel_manager.clone();
+				let min = time_forwardable.as_millis() as u64;
+
+				// TODO: any way we still can use tokio here?
```

Comment:
Yup, will address this in an upcoming "make everything async" PR in which I'll also switch the calls to BDK over to async. Should be noted thought that this is not entirely trivial since the tokio runtime won't be available when we initialize the event handler, so we'll have to pass it around after the fact, i.e., when `start()` gets called and the runtime is setup.

Reply:
And here's the upcoming PR: https://github.com/lightningdevkit/ldk-lite/pull/26